components/server/src/auth/bearer-authenticator.ts

```diff
@@ -103,6 +106,22 @@ export class BearerAuth {
         req.user = user;
     }
 
+    async tryAuthFromHeaders(headers: { [key: string]: any }): Promise<SubjectId | undefined> {
```

Comment:
Just to check we have tests coverage for it? Just be double concious about changing security related code.

Reply:
No, I don't think we have sth yet. But I planned to add test on some level, once I could confirm it working as expected. :+1:

Reply:
Done: [old](https://github.com/gitpod-io/gitpod/pull/19081/commits/2b3d4e6c0e8c7021f1746e9f18f81aa8563001f0) and [new](https://github.com/gitpod-io/gitpod/pull/19081/commits/a8dc1cdd787d75067dd3088b9a88763daebace24)